### PR TITLE
Adds end-to-end (e2e) tests and benchmarks

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds end-to-end (e2e) tests, using the sample data in mosn/mosn and new data for the v2 ABI.

cc @antJack @taoyuanyuan

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
updated with v2 abi as well

File: proxywasm/v2/types.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ const (
 	MetricTypeHistogram MetricType = 3
 )
 
-type MapType int32
+type MapType = int32
```
Comment (codefromthecrypt) [inline proxywasm/v2/types.go:62]: This allows wasmer to use the type alias as a parameter (wazero can use either way)

File: internal/imports/v2/imports.go [hunk 2]
```diff
@@ -50,8 +50,8 @@ func HostFunctions(instance common.WasmInstance) map[string]interface{} {
 
 	hostFuncs["proxy_get_header_map_pairs"] = h.ProxyGetHeaderMapPairs
 
-	hostFuncs["proxy_get_map_values"] = h.ProxyGetMapValues
-	hostFuncs["proxy_set_map_values"] = h.ProxySetMapValues
+	hostFuncs["proxy_get_header_map_value"] = h.ProxyGetHeaderMapValue
```
Comment (codefromthecrypt) [inline internal/imports/v2/imports.go:53]: there are other incompatibilities, but test data doesn't hit them yet. This starts the process of fixing the v2 ABI with real data.

**Comment by codefromthecrypt** [comment]
After this PR, I think we are ready to add wazero!

**Review by antJack — APPROVED** [review]

**Pull request merged**